ydb/core/tx/schemeshard/schemeshard_info_types.cpp

```diff
@@ -19,6 +19,40 @@
 
 #include <util/generic/algorithm.h>
 
+namespace {
+
+using TDiskSpaceQuotas = NKikimr::NSchemeShard::TSubDomainInfo::TDiskSpaceQuotas;
+using TQuotasPair = TDiskSpaceQuotas::TQuotasPair;
+using TStoragePoolUsage = NKikimr::NSchemeShard::TSubDomainInfo::TDiskSpaceUsage::TTables::TStoragePoolUsage;
+
+enum class EDiskUsageStatus {
+    AboveHardQuota,
+    InBetween,
+    BelowSoftQuota,
+};
+
+EDiskUsageStatus CheckStoragePoolsQuotas(const THashMap<TString, TStoragePoolUsage>& storagePoolsUsage,
+                                         const THashMap<TString, TQuotasPair>& storagePoolsQuotas
+) {
+    bool softQuotaExceeded = false;
+    for (const auto& [poolKind, usage] : storagePoolsUsage) {
+        if (const auto* quota = storagePoolsQuotas.FindPtr(poolKind)) {
+            const auto totalSize = usage.DataSize + usage.IndexSize;
+            if (totalSize > quota->HardQuota) {
+                return EDiskUsageStatus::AboveHardQuota;
+            }
+            if (totalSize > quota->SoftQuota) {
```

Comment:
It should be `totalSize >= quota->SoftQuota` for consistency with total quota. For regular quota we require usage to drop _below_ soft quota for the database to become read-write again. You also may need to handle the case where `SoftQuota == 0` (i.e. there is no soft quota and database becomes read-write when not exceeding hard quota).

Reply:
TO DO: change the condition to be `totalSize >= quota->SoftQuota`.

As for the case when the `SoftQuota == 0`, we handle it in the same manner as the regular quotas do, i.e. [we set](https://github.com/jepett0/ydb/blob/a19c3b4dcc28fb1da6d04ecfb139ffdfe90c72fb/ydb/core/tx/schemeshard/schemeshard_info_types.cpp#L102) it to be equal to the `HardQuota` if it wasn't set by the user.